core/domain/platform_parameter_domain.py

```diff
@@ -17,19 +17,46 @@
 """Domain objects for platform parameters."""
 
 from __future__ import annotations
-
+from __future__ import absolute_import
+from __future__ import unicode_literals
 import json
 import re
 
+import enum as enum
+from typing_extensions import TypedDict
+from typing import (
+    Any, Callable, Dict, Iterable, Iterator, List, Optional, Sequence,
+    Type, TypeVar, Tuple, Union, NoReturn)
+
 from core import feconf
 from core import python_utils
 from core import utils
 from core.constants import constants
 from core.domain import change_domain
 
-SERVER_MODES = python_utils.create_enum('dev', 'test', 'prod') # pylint: disable=invalid-name
+# Enum and TypedDict used for MyPy Type annotations
+class SERVER_MODES(enum.Enum):
+    dev = 'dev'
+    test = 'test'
+    prod = 'prod'
+
+class DATA_TYPES(enum.Enum):
+    bool = 'bool'
+    string = 'string'
+    number = 'number'
+
+class Server_context_dict(TypedDict):
+    server_mode: SERVER_MODES
+
+class Filter_dict(TypedDict):
+    type: str
+    conditions: List[Tuple[str,str]]
+
+class PlatformParameterRule_dict(TypedDict):
+    filters: List[Filter_dict]
+    value_when_matched: Any
```

Comment:
we used an Any because the methode value_when_matched return *. line 532 

    `def` value_when_matched(self) -> `Any:` `
       "Returns the value outcome if this rule is matched.

        Returns:
            *. The value outcome.
        "
        return self._value_when_matched`

Reply:
Okay, but then please add an explanatory comment about that.

Reply:
done